test/functional/apps/discover/esql/_esql_view.ts

```diff
@@ -328,8 +332,15 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       describe('with slow queries', () => {
         it('should show only one entry in inspector for table/visualization', async function () {
+          const state = kbnRison.encode({
+            dataSource: { type: 'esql' },
+            query: { esql: 'from kibana_sample_data_flights' },
+          });
+          await PageObjects.common.navigateToActualUrl('discover', `?_a=${state}`, {
+            ensureCurrentUrl: false,
+          });
```

Comment:
@lukasolson I'm not sure exactly why this test started failing, but as far as I can tell it shouldn't have passed originally since `from kibana_sample_data_flights` doesn't show the chart and should only trigger 1 request for `Table`. I suspect we have some race condition (which we should investigate further separately) and the performance improvements cause this scenario to behave as expected, but WDYT?

Reply:
Yeah, you must be right. Here's a recording showing that it was broken before: 

https://github.com/user-attachments/assets/9aff38ea-ec9a-46c6-b374-2f24f505d765

With this fix, it looks like things are working properly.